libraries/AP_FETtecOneWire/AP_FETtecOneWire.h

```diff
@@ -0,0 +1,253 @@
+    */
+    void config_fast_throttle();
+
+#if HAL_WITH_ESC_TELEM
+    /**
+        increment message packet count for every ESC
+    */
+    void inc_send_msg_count();
+
+    /**
+        calculates tx (outgoing packets) error-rate by converting the CRC error counts reported by the ESCs into percentage
+        @param esc_id id of ESC, that the error is calculated for
+        @param esc_error_count the error count given by the esc
+        @return the error in percent
+    */
+    float calc_tx_crc_error_perc(const uint8_t esc_id, uint16_t esc_error_count);
+
+    /**
+        if init is complete checks if the requested telemetry is available.
+        @param t telemetry datastructure where the read telemetry will be stored in.
+        @param centi_erpm 16bit centi-eRPM value returned from the ESC
+        @param tx_err_count Ardupilot->ESC communication CRC error counter
+        @param tlm_from_id receives the ID from the ESC that has respond with its telemetry
+        @return 1 if CRC is correct, 2 on CRC mismatch, 0 on waiting for answer
+    */
+    receive_response decode_single_esc_telemetry(TelemetryData& t, int16_t& centi_erpm, uint16_t& tx_err_count, uint8_t &tlm_from_id);
+#endif
+
+    /**
+        if init is complete sends a single fast-throttle frame containing the throttle for all found OneWire ESCs.
+        @param motor_values a 16bit array containing the throttle values that should be sent to the motors. 0-2000 where 1001-2000 is positive rotation and 0-999 reversed rotation
+        @param tlm_request the ESC to request telemetry from (-1 for no telemetry, 0 for ESC1, 1 for ESC2, 2 for ESC3, ...)
+    */
+    void escs_set_values(const uint16_t *motor_values, const int8_t tlm_request);
+
+    static constexpr uint8_t SERIAL_NR_BITWIDTH = 12;
+
+    typedef struct FETtecOneWireESC
+    {
+        bool active;
+#if HAL_AP_FETTEC_ONEWIRE_GET_STATIC_INFO
+        uint8_t firmware_version;
+        uint8_t firmware_sub_version;
+        uint8_t esc_type;
+        uint8_t serial_number[SERIAL_NR_BITWIDTH];
+#endif
+    } FETtecOneWireESC_t;
+
+    FETtecOneWireESC_t _found_escs[MOTOR_COUNT_MAX]; ///< Zero-indexed array
+    uint32_t _last_config_check_ms;
+#if HAL_WITH_ESC_TELEM
+    float _crc_error_rate_factor; ///< multiply factor. Used to avoid division operations
+    uint16_t _error_count[MOTOR_COUNT_MAX]; ///< error counter from the ESCs. Zero-indexed array
+    uint16_t _error_count_since_overflow[MOTOR_COUNT_MAX]; ///< error counter from the ESCs to pass the overflow. Zero-indexed array
+    uint16_t _send_msg_count; ///< number of fast-throttle commands send by the flight controller
+    uint16_t _update_rate_hz;
+#endif
+    uint16_t _mask;
+    uint8_t _nr_escs_in_bitmask; ///< number of ESCs set on the FTW_MASK parameter
+    uint8_t _found_escs_count;   ///< number of ESCs auto-scanned in the OneWire bus by the scan_escs() function

```

Comment:
This will go away - until all ESCs have moved to RUNNING we don't allow arming sort of thing.

Reply:
this is needed often as well